rlpx.md

```diff
@@ -51,14 +51,14 @@ material for encryption and authentication as
 <code>k<sub>E</sub> || k<sub>M</sub> = KDF(S, 32)</code> as well as a random
 initialization vector `iv`. Alice sends the encrypted message `R || iv || c || d` where
 <code>c = AES(k<sub>E</sub>, iv , m)</code> and
-<code>d = MAC(k<sub>M</sub>, iv || c)</code> to Bob.
+<code>d = MAC(keccak256(k<sub>M</sub>), iv || c)</code> to Bob.
 
 For Bob to decrypt the message `R || iv || c || d`, he derives the shared secret
 <code>S = P<sub>x</sub></code> where
 <code>(P<sub>x</sub>, P<sub>y</sub>) = k<sub>B</sub> * R</code> as well as the encryption and
 authentication keys <code>k<sub>E</sub> || k<sub>M</sub> = KDF(S, 32)</code>. Bob verifies
 the authenticity of the message by checking whether
-<code>d == MAC(k<sub>M</sub>, iv || c)</code> then obtains the plaintext as
+<code>d == MAC(keccak256(k<sub>M</sub>), iv || c)</code> then obtains the plaintext as
```

Comment:
The changes to the ECIES part of the spec are incorrect, we don't hash the mac key using keccak256.

Reply:
I read the code of geth:
https://github.com/ethereum/go-ethereum/blob/a184ab7a61884409a4deea2419c5307a0dc7c49d/crypto/ecies/ecies.go#L274-L276
```go
	Km := K[params.KeyLen:]
	hash.Write(Km)
	Km = hash.Sum(nil)
```

Reply:
You're right :). I checked the code but missed this line. The hash function used is sha256 though.